kbase-extension/static/kbase/js/common/jobManager.js

```diff
@@ -961,6 +972,38 @@ define([
                 return this.getFsmStateFromJobs();
             }
 
+            generateJobParams() {
+                const appParams = [];
+                const inputs = this.model.getItem('inputs'),
+                    params = this.model.getItem('params'),
+                    appSpecs = this.model.getItem('app.specs'),
+                    outputParamIds = this.model.getItem('app.outputParamIds');
+
+                Object.keys(inputs)
+                    .sort()
+                    .forEach((fileType) => {
+                        const appSpecInfo = appSpecs[inputs[fileType].appId].full_info;
+                        params[fileType].filePaths.forEach((filePathParams) => {
+                            const filteredParams = {
+                                app_id: appSpecInfo.id,
+                            };
+                            filteredParams.userParams = {};
+                            appSpecs[appSpecInfo.id].parameters.forEach((param) => {
+                                const paramId = param.id;
+                                if (filePathParams[paramId]) {
+                                    filteredParams.userParams[paramId] = filePathParams[paramId];
+                                } else if (params[fileType].params[paramId]) {
```

Comment:
`filePathParams` are unique per job, and the `params[fileType].params` are shared between all of an app's jobs